File: backend/app/schemas/image.py
"""
[IDENTITY]: Image Generation Schemas
DTOs for AI Image Generation interactions.

[INPUT]:
- ImageGenerationRequest (Style, Category, Asset, Product).

[LINK]:
- ImageRouter -> ../api/v1/endpoints/image.py
- ImageModel -> ../models/image.py

[OUTPUT]: ImageGenerationResponse (Async Task ID).
[POS]: /backend/app/schemas/image.py

[PROTOCOL]:
1. `reference_image_id` is optional; serves as "img2img" input source.
"""

from typing import Optional, List
from uuid import UUID
from datetime import datetime

from pydantic import BaseModel, Field

# Import enums from models to avoid duplication (DRY)
from app.models.image import StyleType, JobStatus
from app.models.product import ProductCategory


class ImageGenerationRequest(BaseModel):
    """Request schema for image generation endpoint."""
    style_id: StyleType = Field(
        ...,
        description="Visual style for generation"
    )
    category_id: ProductCategory = Field(
        ...,
        description="Product category from previous step"
    )
    asset_id: UUID = Field(
        ...,
        description="Original asset ID"
    )
    product_id: UUID = Field(
        ...,
        description="Product ID from category selection step"
    )
    # Story 2.4: Reference image attachment
    reference_image_id: Optional[UUID] = Field(
        default=None,
        description="Optional reference image ID to guide generation"
    )


class ImageGenerationResponse(BaseModel):
    """Response schema for image generation request (202 Accepted)."""
    task_id: UUID = Field(..., description="Celery task ID for status polling")
    status: JobStatus = Field(default=JobStatus.PENDING)
    message: Optional[str] = Field(
        default="Generation task queued successfully",
        description="User-friendly status message"
    )

    model_config = {"from_attributes": True}


class JobStatusResponse(BaseModel):
    """Response schema for job status query."""
    task_id: UUID
    status: JobStatus
    progress: int = Field(default=0, ge=0, le=100)
    error_message: Optional[str] = None
    result_urls: Optional[List[str]] = None
    created_at: datetime
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class GenerationErrorResponse(BaseModel):
    """Error response schema for generation failures."""
    detail: str
    error_code: str

File: frontend/src/types/image.ts
/**
 * === HIGH-INTEGRITY HEADER ===
 *
 * [IDENTITY]: Image Generation Type Definitions
 * Frontend type definitions for AI image generation feature.
 *
 * Story关联: Story 2.1 - Style Selection & Generation Trigger
 *
 * [INPUT]: N/A (Type Definition File)
 *
 * [LINK]:
 * - Backend Schemas -> backend/app/schemas/image.py
 * - API Client -> @/lib/api/images.ts
 * - Dependency -> @/types/product.ts
 *
 * [OUTPUT]:
 * - StyleType, GenerationParams, GenerationResponse, JobStatusResponse
 *
 * [POS]: /frontend/src/types/image.ts
 *
 * [PROTOCOL]:
 * 1. Exported types must match Backend Pydantic models.
 * 2. Field names converted: snake_case (Backend) -> camelCase (Frontend).
 * 3. Type changes require recursive updates in components and API clients.
 *
 * === END HEADER ===
 */

/**
 * Visual style types for image generation
 */
export type StyleType = 'modern' | 'luxury' | 'fresh' | 'tech' | 'warm' | 'business';

/**
 * Generation job status
 */
export type GenerationStatus = 'idle' | 'pending' | 'processing' | 'completed' | 'failed';

/**
 * Style option for UI display
 */
export interface StyleOption {
    id: StyleType;
    name: string;
    gradient: string;
    description?: string;
}

/**
 * Parameters for image generation request
 */
export interface GenerationParams {
    styleId: StyleType;
    categoryId: string;
    assetId: string;
    productId: string;
}

/**
 * Response from image generation API
 */
export interface GenerationResponse {
    taskId: string;
    status: GenerationStatus;
    message?: string;
}

/**
 * Job status response for polling
 */
export interface JobStatusResponse {
    taskId: string;
    status: GenerationStatus;
    progress: number;
    errorMessage?: string;
    resultUrls?: string[];
    createdAt: string;
    startedAt?: string;
    completedAt?: string;
}

/**
 * Default style options with CSS gradients
 * Based on UX specifications in Story 2.1
 */
export const STYLE_OPTIONS: StyleOption[] = [
    {
        id: 'modern',
        name: '现代',
        gradient: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
        description: '简约、干净的现代风格'
    },
    {
        id: 'luxury',
        name: '奢华',
        gradient: 'linear-gradient(135deg, #f093fb 0%, #f5576c 100%)',
        description: '高端、精致的奢华感'
    },
    {
        id: 'fresh',
        name: '清新',
        gradient: 'linear-gradient(135deg, #fa709a 0%, #fee140 100%)',
        description: '活力、年轻的清新感'
    },
    {
        id: 'tech',
        name: '科技',
        gradient: 'linear-gradient(135deg, #30cfd0 0%, #330867 100%)',
        description: '未来感、科技感'
    },
    {
        id: 'warm',
        name: '温暖',
        gradient: 'linear-gradient(135deg, #ff9a9e 0%, #fecfef 100%)',
        description: '温馨、舒适的暖色调'
    },
    {
        id: 'business',
        name: '商务',
        gradient: 'linear-gradient(135deg, #a8edea 0%, #fed6e3 100%)',
        description: '专业、可信赖的商务风'
    }
];
